service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CopyGcsBucketDefinitionStep.java

```diff
@@ -23,45 +23,35 @@
 import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ResourceKeys;
 import java.util.Optional;
 import java.util.UUID;
+import javax.annotation.Nullable;
 import org.springframework.http.HttpStatus;
 
 public class CopyGcsBucketDefinitionStep implements Step {
 
   private final AuthenticatedUserRequest userRequest;
   private final ControlledGcsBucketResource sourceBucket;
   private final ControlledResourceService controlledResourceService;
+  private final CloningInstructions resolvedCloningInstructions;
 
   public CopyGcsBucketDefinitionStep(
       AuthenticatedUserRequest userRequest,
       ControlledGcsBucketResource sourceBucket,
-      ControlledResourceService controlledResourceService) {
+      ControlledResourceService controlledResourceService,
+      CloningInstructions resolvedCloningInstructions) {
     this.userRequest = userRequest;
     this.sourceBucket = sourceBucket;
     this.controlledResourceService = controlledResourceService;
+    this.resolvedCloningInstructions = resolvedCloningInstructions;
   }
 
   @Override
   public StepResult doStep(FlightContext flightContext)
       throws InterruptedException, RetryException {
     final FlightMap inputParameters = flightContext.getInputParameters();
     final FlightMap workingMap = flightContext.getWorkingMap();
-    final CloningInstructions cloningInstructions =
-        Optional.ofNullable(
-                inputParameters.get(
-                    ControlledResourceKeys.CLONING_INSTRUCTIONS, CloningInstructions.class))
-            .orElse(sourceBucket.getCloningInstructions());
-    // future steps need the resolved cloning instructions
-    workingMap.put(ControlledResourceKeys.CLONING_INSTRUCTIONS, cloningInstructions);
-    if (CloningInstructions.COPY_NOTHING.equals(cloningInstructions)) {
-      final ApiClonedControlledGcpGcsBucket noOpResult =
-          new ApiClonedControlledGcpGcsBucket()
-              .effectiveCloningInstructions(cloningInstructions.toApiModel())
-              .bucket(null)
-              .sourceWorkspaceId(sourceBucket.getWorkspaceId())
-              .sourceResourceId(sourceBucket.getResourceId());
-      FlightUtils.setResponse(flightContext, noOpResult, HttpStatus.OK);
-      return StepResult.getStepResultSuccess();
-    } // todo: handle COPY_REFERENCE PF-811, PF-812
+    FlightUtils.validateRequiredEntries(inputParameters,
+        ControlledResourceKeys.DESTINATION_WORKSPACE_ID);
+     // todo: handle COPY_REFERENCE PF-811, PF-812
```

Comment:
I think the comment is obsolete. PF-811 is DONE and PF-812 is CLOSED

Reply:
Yeah, nobody has asked for clone-resource-to-reference; it was just implied by the structure of the cloning instructions as I read it.

Reply:
Why leave the comment, though?